src/util/handlers/PrimaryHandler.ts

```diff
@@ -0,0 +1,27 @@
+import type { ClusterManager } from '../../init/ClusterManager';
+import { InternalServerError } from '../errors/InternalServerError';
+import type { AsyncHandler } from './AsyncHandler';
+import { SequenceHandler } from './SequenceHandler';
+
+/**
+ * A composite handler that will only run all supporting handlers sequentially
+ * if they are executed from the primary process and return the value of the
+ * last supported handler.
+ * The `canHandle` check of this handler will always succeed if single threaded,
+ * or if run from the primary process.
+ */
+export class PrimaryHandler<TIn = void, TOut = void> extends SequenceHandler<TIn, TOut | undefined> {
+  private readonly clusterManager: ClusterManager;
+
+  public constructor(handlers: AsyncHandler<TIn, TOut>[], clusterManager: ClusterManager) {
+    super(handlers);
+    this.clusterManager = clusterManager;
+  }
+
+  public async canHandle(): Promise<void> {
+    if (this.clusterManager.isSingleThreadedOrPrimary()) {
+      return Promise.resolve();
+    }
+    return Promise.reject(new InternalServerError('Will not execute on worker process.'));
```

Comment:
```suggestion
    if (!this.clusterManager.isSingleThreadedOrPrimary()) {
      throw new NotImplementedHttpError('Will not execute on worker process.');
    }
```
`NotImplementedHttpError` is the error we usually use if a class says it does not care about the request, but there is nothing specifically wrong.